### DISCUSSION TITLE
Raspberry Pi harvester stuck reconnecting to host

### DISCUSSION THREAD

**Discussion by Zeesy** [Chia-Network/chia-blockchain#5889] [Support]
I'm trying to set up a raspberry pi as a harvester, but am struggling to get it connected to the full node running off my Windows 10 machine. [This thread](https://github.com/Chia-Network/chia-blockchain/issues/4824) had a similar problem, but I am not getting the same results from the solution. I've altered config.yaml, but the results I get from cat ~/.chia/mainnet/log/debug.log are:

`2021-05-23T13:27:07.228 daemon chia.daemon.server : INFO Starting Daemon Server 2021-05-23T13:27:07.239 daemon chia.daemon.server : INFO Waiting Daemon WebSocketServer closure 2021-05-23T13:27:12.037 harvester harvester : INFO Started harvester service on network_id: mainnet 2021-05-23T13:27:12.049 harvester harvester : INFO Reconnecting to peer {'host': '192.168.0.10', 'port': 8447} 2021-05-23T13:27:12.109 daemon chia.daemon.server : INFO Register service {'service': 'chia_harvester'} 2021-05-23T13:27:12.111 daemon chia.daemon.server : INFO registered for service chia_harvester 2021-05-23T13:27:12.112 daemon chia.daemon.server : INFO {'success': True} 2021-05-23T13:27:45.571 harvester harvester : INFO Reconnecting to peer {'host': '192.168.0.10', 'port': 8447} 2021-05-23T13:28:19.543 harvester harvester : INFO Reconnecting to peer {'host': '192.168.0.10', 'port': 8447}`

Then reconnecting all the way down. The results should be:
`2021-05-22T21:54:02.898 harvester chia.harvester.harvester: INFO [number] plots were eligible for farming [key] Time: 0.23380 s. Total x plots`

In the GUI on the main machine, only the local plots passed filter are visible, unlike the scenario I'm expecting from [this tutorial](https://medium.com/geekculture/how-to-use-your-raspberry-pi-as-a-chia-harvester-66440c17d318). What am I missing here?

**Comment by stefano-gentile** [comment]
Just in PI config.yaml you should replace this

```
harvester:
  chia_ssl_ca:
    crt: config/ssl/ca/chia_ca.crt
    key: config/ssl/ca/chia_ca.key
  farmer_peer:
    host: Main.Machine.IP
    port: 8447
```

Main Machine IP is your W10 main pc IPV4 address, check that it is well written. You don't need to make changes on W10 config.yaml.

**Reply by Zeesy** [reply]
Main.Machine.IP is set correctly to the W10 IPv4 address, but the harvester is stuck reconnecting.

**Reply by Zeesy** [reply]
any updates on this?

**Reply by loppefaaret** [reply]
can you reach the port 8447 port on the remote machine with telnet from the harvesting machine ?

**Reply by Zeesy** [reply]
An update from here: it looks like I'm stuck behind a CGNAT, which is blocking port 8444. Some info here:
https://www.reddit.com/r/chia/comments/ne7na4/cgnat_and_cant_open_port_8444/

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
